### PR TITLE
tests(perf): add perf scenario for sync api with apikey plan 

### PR DESCRIPTION
**Pull request by ytvnr**
## Issue

https://gravitee.atlassian.net/browse/APIM-621

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ykuqrnqniv.chromatic.com)

### CODE REVIEW

File: gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  });
+  failIf(changeLifecycleResponse.status !== 204, 'Could not change lifecycle');
+
+  GatewayClient.waitForApiAvailability({ contextPath: api.contextPath });
+
+  return { api: createdApi, plan: createdPlan, waitGateway: { contextPath: api.contextPath } };
+}
+
+export default (data: GatewayTestData) => {
+  const res = http.get(k6Options.apim.gatewayBaseUrl + data.waitGateway.contextPath);
```
Comment (leleueri) [inline gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts]: APIKey is not set to the request.

In addition subscriptions are not managed, we should at least have a subscription a the end of the setup phase.
In the ideal case, we should be able to define the number of subscriptions (so the number of APIKeys) and peek randomly an ApiKey in the simulation.

File: gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  check(res, {
+    'status is 200': () => res.status === 200,
+    'contains header': () => res.headers['Content-Type'] === 'application/xml;charset=UTF-8',
+  });
+};
+
+export function teardown(data: GatewayTestData) {
+  ApisClient.changeLifecycle(data.api.id, LifecycleAction.STOP, {
```
Comment (leleueri) [inline gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts:176]: Remember to unsubscribe plan here before cleaning everything

File: gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      'Content-Type': 'application/json',
+      ...authorizationHeaderFor(ADMIN_USER),
+    },
+  });
+  failIf(apiCreationResponse.status !== 201, 'Could not create API');
+  const createdApi = HttpHelper.parseBody<ApiEntity>(apiCreationResponse);
+
+  const planCreationResponse = PlansClient.createPlan(
+    createdApi.id,
+    PlansFixture.newPlan({ status: PlanStatus.PUBLISHED, security: PlanSecurityType.API_KEY }),
```
Comment (leleueri) [inline gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts]: can we force the auto approval to be explicit ?

**Review by leleueri — CHANGES_REQUESTED** [review]
Subscriptions on ApiKey plan are not managed.
I comment only one file but same comment can be applied to all ApiKey simulations

File: gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts (hunk 1 above)
Comment (ytvnr) [reply gravitee-apim-perf/src/scenarios/apikey-api-body-policies/apikey.v3.test.ts]: good catch :)

**Review by leleueri — APPROVED** [review]

**Pull request merged**